gst/nnstreamer/nnstreamer_plugin_api_impl.c

```diff
@@ -1428,3 +1428,91 @@ gst_tensor_meta_info_append_header (GstTensorMetaInfo * meta, GstMemory * mem)
   gst_memory_unmap (new_mem, &new_map);
   return new_mem;
 }
+
+#define NNS_TENSOR_EXTRA_MAGIC 0xf00dc0de
+
+/**
+ * @brief TBU
+*/
+void
+gst_tensors_extra_init (GstTensorExtraInfo * extra)
+{
+  guint i;
+
+  extra->magic = NNS_TENSOR_EXTRA_MAGIC;
+  extra->version = 0;
+  extra->num_extra_tensors = 0;
+  for (i = 0; i < NNS_TENSOR_SIZE_EXTRA_LIMIT; ++i) {
+    gst_tensor_info_init (&extra->infos[i]);
+  }
+}
+
+/**
+ * @brief TBU
+*/
+GstMemory *
+gst_tensors_get_nth_memory (GstBuffer * buffer, guint index)
+{
+  guint buffer_n_memory, i, offset = 0;
+  GstMemory *extra_tensors_memory, *res_mem;
+  GstMapInfo extra_tensors_map;
+  GstTensorExtraInfo *extra_info;
+
+  buffer_n_memory = gst_buffer_n_memory (buffer);
```

Comment:
If the input buffer is not GstBuffer, the buffer_n_memory is 0.
and gst_buffer_get_memory return 'null' if buffer is not GstBuffer;

How about add below to line 1461 ?
g_return_val_if_fail (GST_IS_BUFFER (buffer), NULL);

Reply:
Thanks for the suggestion, I updated the code.